caluma/caluma_form/filters.py

```diff
@@ -339,9 +343,26 @@ def filter(self, qs, value):
             qs = self._apply_filter(qs, val)
         return qs
 
-    def _apply_filter(self, qs, value):
+    def _get_questions(self, value):
+        raw_questions = questions = value.get("questions")
+        raw_forms = value.get("forms")
+        if not raw_questions and not raw_forms:
+            raise exceptions.ValidationError(
+                '"forms" and/or "questions" parameter must be set'
+            )
 
-        questions = self._validate_and_get_questions(value["questions"])
+        if raw_forms:
+            forms = self._validate_and_get_forms(raw_forms)
+            questions = Form.get_all_questions(forms).exclude(
+                type__in=self.UNSUPPORTED_QUESTION_TYPES
+            )
```

Comment:
Wouldn't it be a good idea to validate the questions list and reject the call if the client supplies unsupported questions? Otherwise, we might get unexpected behaviour

Reply:
refactored to check before, so the exclude isnt needed